src/pages/home/report/FloatingMessageCounter/index.js

```diff
@@ -28,73 +26,67 @@ const defaultProps = {
 const MARKER_INACTIVE_TRANSLATE_Y = -40;
 const MARKER_ACTIVE_TRANSLATE_Y = 10;
 
-class FloatingMessageCounter extends PureComponent {
-    constructor(props) {
-        super(props);
-        this.translateY = new Animated.Value(MARKER_INACTIVE_TRANSLATE_Y);
-        this.show = this.show.bind(this);
-        this.hide = this.hide.bind(this);
-    }
-
-    componentDidUpdate() {
-        if (this.props.isActive) {
-            this.show();
-        } else {
-            this.hide();
-        }
-    }
+function FloatingMessageCounter(props) {
+    const {translate} = useLocalize();
+    const translateY = useMemo(() => new Animated.Value(MARKER_INACTIVE_TRANSLATE_Y), []);
 
-    show() {
-        Animated.spring(this.translateY, {
+    const show = useCallback(() => {
+        Animated.spring(translateY, {
             toValue: MARKER_ACTIVE_TRANSLATE_Y,
             duration: 80,
             useNativeDriver: true,
         }).start();
-    }
+    }, [translateY]);
```

Comment:
`translateY` will never change, right? Why is it in the dependency array?

Reply:
When running `npm run lint` it gives me a warning that `useCallback` is missing a dependency `translateY`. It suggests adding it or removing the dependency array. When I remove the dependency array and run `lint` again it tells me to add an array. I kept `translateY` as a dependency to keep `lint` happy 😅.

Would it be best to use something else other than `useCallback`

Reply:
no, this is fine the.